### PR TITLE
modifying view single event

### PR DESCRIPTION
**Pull request by idosavion**
Added button for event owner to change pending event status
Add functionality for DataStreamer

### CODE REVIEW

File: app/src/main/java/com/huji/foodtricks/buddies/EventAttendanceProvider.java [hunk 1]
```diff
@@ -22,6 +20,19 @@
     private HashMap<String, String> tentatives;
     private HashMap<String, String> nonResponsive;
 
+    public enum RSVP {ATTENDING, NOT_ATTENDING, TENTATIVE, NON_RESPONSIVE};
+
+    public RSVP getUserRSVP(String userId) {
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/EventAttendanceProvider.java:22]: a. this should appear later in the code (after the constructor, at least)
b. it's ok but i would write it like this:

```
    public RSVP getUserRSVP(String userId) {
        if (attending.containsKey(userId)) {
            return RSVP.ATTENDING;
        }
        if (tentatives.containsKey(userId)) {
            return RSVP.TENTATIVE;
        }
        if (notAttending.containsKey(userId)) {
            return RSVP.NOT_ATTENDING;
        }
        if (nonResponsive.containsKey(userId)) {
            return RSVP.NON_RESPONSIVE;
        }
        return null;
    }
```

(single-line if's should still have curly braces)

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 2]
```diff
@@ -32,6 +31,7 @@
 
 public class ViewSingleEventActivity extends AppCompatActivity {
 
+    public static final HashSet<Integer> ALL_RSVP_BUTTONS = new HashSet<>(Arrays.asList(R.id.approve_btn, R.id.tentative_btn, R.id.decline_btn));
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java]: ->
```
    public static final HashSet<Integer> ALL_RSVP_BUTTONS = 
            new HashSet<>(Arrays.asList(R.id.approve_btn, R.id.tentative_btn, R.id.decline_btn));
```

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 3]
```diff
@@ -142,20 +141,38 @@ public void onUpdateSuccess() {
...
+        EventAttendanceProvider attendanceProvider = curr_event.getAttendanceProvider();
+        EventAttendanceProvider.RSVP status = attendanceProvider.getUserRSVP(currentUserID);
+        if (status == EventAttendanceProvider.RSVP.ATTENDING) {
+            changeButtonToEnabled(findViewById(R.id.approve_btn));
+        }
+        else if (status == EventAttendanceProvider.RSVP.NOT_ATTENDING)
+            changeButtonToEnabled(findViewById(R.id.decline_btn));
+        else if (status == EventAttendanceProvider.RSVP.TENTATIVE)
+            changeButtonToEnabled(findViewById(R.id.tentative_btn));
+
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java]: remove extra empty lines

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 4]
```diff
@@ -142,20 +141,38 @@ public void onUpdateSuccess() {
...
+    }
 
-        for (int buttonId : allRSVPButtons) {
-            Button currButton = findViewById(buttonId);
-            currButton.setBackgroundColor(Color.WHITE);
-            currButton.setTextColor(Color.BLACK);
-        }
+    private void changeButtonToDisabled(Button currButton) {
+        currButton.setBackgroundColor(Color.WHITE);
+        currButton.setTextColor(Color.BLACK);
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java:183]: you should also actually disable the button (and enable it in "changeToEnabled") so the user can't click it again..

File: app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java [hunk 5]
```diff
@@ -262,4 +281,30 @@ private void addEventIdToUser(UserModel userModel, String userModelId, String ev
         userModel.addChangedEvent(eventId);
         modifyUser(userModel, userModelId, completion);
     }
+    public void fetchAllUserModels(UsersArrayListFetchingCompletion completion) {
+        DatabaseReference ref = mDatabase.child("users");
+        ref.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                Iterable<DataSnapshot> users = dataSnapshot.getChildren();
+//                HashMap<String, String> usersMap = new HashMap<>();
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/DatabaseStreamer.java]: remove

File: app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java [hunk 6]
```diff
@@ -54,6 +76,8 @@ protected void onCreate(Bundle savedInstanceState) {
 
         Objects.requireNonNull(getSupportActionBar()).setTitle(curr_event.getTitle());
         updateAllFields(curr_event);
+
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/buddies/ViewSingleEventActivity.java:57]: remove extra empty lines

**Review by mfocshaner — DISMISSED** [review]
Hey, great job adding all this;
I added a few comments - stuff to clean/reorder before A&M's code review.

GJ! 100/100 <3

File: app/src/main/java/com/huji/foodtricks/buddies/EventAttendanceProvider.java (hunk 1 above)
Comment (idosavion) [reply app/src/main/java/com/huji/foodtricks/buddies/EventAttendanceProvider.java:22]: fixed

**Comment by idosavion** [comment]
Fixed all comments - please reapprove and send mesos

**Review by mfocshaner — APPROVED** [review]
Good job

**Pull request merged**